pkg/metricsproxy/countervec.go

```diff
@@ -44,12 +48,8 @@ func NewCounterVec(opts prometheus.CounterOpts, labelNames []string) *CounterVec
 //     myVec.WithLabelValues("404", "GET").Add(42)
 func (c *CounterVecProxy) WithLabelValues(lvs ...string) prometheus.Counter {
 	if len(lvs) > 0 {
-		labels := make(map[string]string, len(lvs))
-		for index, label := range lvs {
-			labels[c.LabelNames[index]] = label
-		}
 		c.mu.Lock()
-		noteLabelsInMetricsProxy(c, labels)
+		noteLabelsInMetricsProxy(c, lvs)
```

Comment:
How about only lock `c.Labels` (not lock two-level function call and `strings.Join`)? or maybe you can try `sync.Map`.